src/ui/VirtualFileRenderer/VirtualFileRenderer.spec.tsx

```diff
@@ -0,0 +1,361 @@
+      dateNowSpy
+        .mockImplementationOnce(() => 1000)
+        .mockImplementationOnce(() => 2000)
+      requestAnimationFrameSpy.mockImplementation((cb) => {
+        setTimeout(() => {
+          cb()
+        }, 50)
+        return 1
+      })
+
+      render(
+        <VirtualFileRenderer
+          code={code}
+          coverage={coverageData}
+          fileName="tsx"
+        />,
+        {
+          wrapper: wrapper(),
+        }
+      )
+
+      const lines = await screen.findAllByText(
+        /{ pageName: 'repo', text: repo },/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      const codeRenderer = screen.getByTestId('virtual-file-renderer')
+      await waitFor(() =>
+        expect(codeRenderer).toHaveStyle('pointer-events: none')
+      )
+    })
+
+    it('reset pointer events after scroll', async () => {
+      dateNowSpy
+        .mockImplementationOnce(() => 1000)
+        .mockImplementationOnce(() => 2000)
+      requestAnimationFrameSpy.mockImplementation((cb) => {
+        setTimeout(() => {
+          cb()
+        }, 50)
+        return 1
+      })
+
+      render(
+        <VirtualFileRenderer
+          code={code}
+          coverage={coverageData}
+          fileName="tsx"
+        />,
+        {
+          wrapper: wrapper(),
+        }
+      )
+
+      const lines = await screen.findAllByText(
+        /{ pageName: 'repo', text: repo },/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      const codeRenderer = screen.getByTestId('virtual-file-renderer')
+      await waitFor(() =>
+        expect(codeRenderer).toHaveStyle('pointer-events: none')
+      )
+      await waitFor(() =>
+        expect(codeRenderer).toHaveStyle('pointer-events: auto')
+      )
+    })
+
+    it('calls cancelAnimationFrame', async () => {
+      dateNowSpy
+        .mockImplementationOnce(() => 1000)
+        .mockImplementationOnce(() => 2000)
+      requestAnimationFrameSpy.mockImplementation((cb) => {
+        setTimeout(() => {
+          cb()
+        }, 50)
+        return 1
+      })
+
+      const { container } = render(
+        <VirtualFileRenderer
+          code={code}
+          coverage={coverageData}
+          fileName="tsx"
+        />,
+        {
+          wrapper: wrapper(),
+        }
+      )
+
+      const lines = await screen.findAllByText(
+        /{ pageName: 'repo', text: repo },/
+      )
+      expect(lines[0]).toBeInTheDocument()
+
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      // eslint-disable-next-line testing-library/no-container
+      container.remove()
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+      await fireEvent.scroll(window, { target: { scrollX: 100 } })
+
+      await waitFor(() => expect(cancelAnimationFrameSpy).toHaveBeenCalled())
+    })
+  })
```

Comment:
Not really sure what you are trying to test here? Is there something more tangible that we can assert besides that cancelAnimationFrame is called?